src/JsonApiSerializer/JsonConverters/ResourceObjectConverter.cs

```diff
@@ -321,5 +321,16 @@ protected virtual string GenerateDefaultTypeName(Type type)
         {
             return type.Name.ToLowerInvariant();
         }
+
+        /// <summary>
+        /// Exposes contract to allow overriding object initialisation based on resource type during deserialization.
+        /// </summary>
+        /// <param name="contract"></param>
+        /// <param name="type"></param>
+        /// <returns></returns>
+        protected virtual object CreateDefault(JsonObjectContract contract, string type)
+        {
+            return contract.DefaultCreator();

```

Comment:
This creates null reference if the type is an interface. Although this was existing behaviour it makes more sense to check if defaultCreator == null and throw a meaningful exception